[refactor] search.js: use custom auto completion implementation

Bnyro opened the pull request:
## What does this PR do?
- refactor: use custom auto completion implementation

## Why is this change important?
- the previously used library is unmaintained for 6 years now, see https://github.com/searxng/searxng/pull/4284#discussion_r1954493434

## How to test this PR locally?
- enable an autocompletion provider
- test different HTTP methods from the privacy settings
- test selecting suggestions with the arrow keys
- test pressing enter on the search form
- test inserting garbage queries that don't return any autocompletion suggest

## Related issues
see https://github.com/searxng/searxng/pull/4284#discussion_r1954493434

vojkovic commented on searx/static/themes/simple/src/js/main/search.js:
```diff
@@ -35,118 +32,105 @@
+
+          setTimeout(() => {
+            if (query == qinput.value) fetchResults(query);
+          }, 300);
```
If you don't mind me asking, how come you are adding a delay to the autocomplete? It seems to make the user experience of using autocomplete much worse.

With autocomplete.js, there was a bug where if the previous request hadn't finished, the next request would cancel and never get sent (so a delay of 300ms was added which would make the issue less apparent).

Let me know your thoughts.

Bnyro replied:
> If you don't mind me asking, how come you are adding a delay to the autocomplete? It seems to make the user experience of using autocomplete much worse.
>
The goal of this PR is to rebuild the autocomplete behavior of autocomplete.js without any changes, thus I kept the delay.

> With autocomplete.js, there was a bug where if the previous request hadn't finished, the next request would cancel and never get sent (so a delay of 300ms was added which would make the issue less apparent).
>
Interesting, I wasn't aware of that.

Nevertheless I think such behavior changes should be discussed in an other issue, as I said the only goal of this PR is to remove the very outdated, unmaintained `autocomplete.js` dependency without changing any behavior.

Bnyro committed:
[refactor] search.js: use custom auto completion implementation

The previously used library is unmaintained for 6 years now [1] and the solution
had know issues [2][3]

[1] https://github.com/searxng/searxng/pull/4284#discussion_r1954493434
[2] https://github.com/searxng/searxng/pull/4318#issuecomment-2731576657
[3] https://github.com/privau/searxng/issues/56

return42 committed:
[build] /static

return42 reviewed (approved):
I slightly modified the commit message / added ... to the body:

```
The previously used library is unmaintained for 6 years now [1] and the solution
had know issues [2][3]

[1] https://github.com/searxng/searxng/pull/4284#discussion_r1954493434
[2] https://github.com/searxng/searxng/pull/4318#issuecomment-2731576657
[3] https://github.com/privau/searxng/issues/56
```

and I pushed the commit from `make static.build.commit` on top.

@vojkovic @Bnyro from my side this PR is ready to merge / do you have any open questions?
